src/precice/SolverInterface.hpp

```diff
@@ -361,6 +361,19 @@ class SolverInterface {
    */
   bool isMeshConnectivityRequired(int meshID) const;
 
+  /**
+   * @brief Checks if the given data set requires gradient data.
+   * We check if the data object has been intialized with the gradient flag.
+   *
+   * preCICE may require gradient data information from the solver and
+   * ignores any API calls regarding gradient data if it is not required.
+   * (When applying a nearest-neighbor-gradient mapping)
+   *
+   * @param[in] dataID the id of the data
+   * @returns whether gradient is required
+   */
+  bool isDataGradientRequired(int dataID) const;
```

Comment:
```suggestion
  bool isGradientRequired(int dataID) const;
```
? Could be sufficient already as we pass a dataID?
No strong opinion, just a suggestion.